### PR TITLE
add more checks for keylen

### PR DESCRIPTION
**Pull request by fanatid**
Part of #22

### CODE REVIEW

**Comment by fanatid** [comment]
```shell
$ node -v
v5.6.0
$ node
> require('crypto').pbkdf2Sync('password', 'salt', 1, NaN)
TypeError: Bad key length
$ nvm use 0.10
$ node -v
v0.10.42
$ node
> require('crypto').pbkdf2Sync('password', 'salt', 1, NaN)
<SlowBuffer >
```

Different behavior for different node versions :(

File: test/index.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ var compatNode = require('../')
 var compatBrowser = require('../browser')
 var fixtures = require('./fixtures')
 
+// put NaN and Infinity
+fixtures.invalid[fixtures.invalid.length - 2].dkLen = NaN
+fixtures.invalid[fixtures.invalid.length - 1].dkLen = Infinity
```
Comment (dcousens) [inline test/index.js]: I'd rather add these fixtures inline here,  then do this weird modification which is error prone for others adding tests.

Comment (fanatid) [inline test/index.js]: Agree, any ideas how tests dkLen with `NaN` and `Infinity`?

Comment (dcousens) [inline test/index.js]: What do you mean?

Comment (fanatid) [inline test/index.js]: oops, sorry

**Comment by fanatid** [comment]
updated

**Comment by dcousens** [comment]
... all failed except Node 4? Interesting

**Comment by dcousens** [comment]
``` javascript
 1) node pbkdf2 pbkdf2-compat pbkdf2Sync sha1 should throw Bad key length:
     AssertionError: Missing expected exception..
      at Function._throws (assert.js:301:5)
      at Function.assert.throws (assert.js:318:11)
      at Context.<anonymous> (test/index.js:95:24)
```

**Comment by fanatid** [comment]
What version of node v4 you are using? May be your version not include https://github.com/nodejs/node/commit/4c8d96bc30281d77ab556adfe9cc1f91e0db4522 ? (checks on `NaN` and `Infinity` was added there)

**Comment by dcousens** [comment]
@fanatid I'm referring to the CI.
Every version **except** Node 4 are failing this.

**Comment by fanatid** [comment]
@dcousens I'll fix with this PR after https://github.com/nodejs/node/pull/5397 will be merged and released

**Comment by calvinmetcalf** [comment]
merged as part of #27

**Pull request closed without merging**